File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/ServerBot.java
package io.github.alexiscomete.lapinousecond.worlds;

import io.github.alexiscomete.lapinousecond.save.CacheGetSet;
import io.github.alexiscomete.lapinousecond.save.Tables;

/**
 * Représente un serveur discord dans la base de données
 */
public class ServerBot extends CacheGetSet {

    /**
     * Créer un nouveau serveur / ! \ passer par SaveManager
     * @param id l'identifiant discord du serveur
     */
    public ServerBot(long id) {
        super(id, Tables.SERVERS.getTable());
    }

    /**
     * Permet de tester si le paramètre est valid en nombre de caractères puis de le régler
     * @param len longueur du message
     * @param message la valeur du paramètre
     * @param prog_name le paramètre
     * @return un message d'erreur s'il y a trop de caractères, il faudrait recoder cette partie
     */
    public String testValueAndSet(int len, String message, String prog_name) {
        if (message.length() < len) {
            set(prog_name, message);
            return "";
        } else {
            return "Impossible : trop de caractères : " + message.length() + ", nombre autorisé : " + len;
        }
    }
}

File: src/main/java/io/github/alexiscomete/lapinousecond/worlds/ServerBot.kt
package io.github.alexiscomete.lapinousecond.worlds

import io.github.alexiscomete.lapinousecond.data.managesave.CacheCustom
import io.github.alexiscomete.lapinousecond.data.managesave.CacheGetSet
import io.github.alexiscomete.lapinousecond.data.managesave.Table

val SERVERS = Table("guilds")
val servers = CacheCustom(SERVERS) { id: Long -> ServerBot(id) }

/**
 * Représente un serveur discord dans la base de données
 * Pour créer un serveur utilisez le CacheCustom
 * @param id l'identifiant discord du serveur
 */
class ServerBot(id: Long) : CacheGetSet(id, SERVERS) {

    fun addPlace(id: Long) {
        val places = this["places"]
        if (places == "") {
            this["places"] = id.toString()
        } else {
            this["places"] += ",$id"
        }
    }

    fun removePlace(id: Long) : Boolean {
        val places = this["places"]
        return if (places == "") {
            false
        } else if (places.startsWith("$id,")) {
            this["places"] = places.replace("$id,", "")
            true
        } else if (places == id.toString()) {
            this["places"] = ""
            true
        } else {
            this["places"] = places.replace(",$id,", ",")
            true
        }
    }

    fun getPlaces() : List<Long> {
        val places = this["places"]
        return if (places == "") {
            listOf()
        } else {
            places.split(",").map { it.toLong() }
        }
    }
}